dart/README.md

```diff
@@ -0,0 +1,219 @@
+# Fury Dart
+
+## Overview
+
+This PR adds Dart language support to Apache Fury, implementing a comprehensive serialization solution for Dart and Flutter applications. Fury Dart consists of approximately 15,000 lines of code and provides an efficient serialization mechanism that works within Flutter's reflection limitations.
+
+## Implementation Approach
+
+Dart supports reflection, but Flutter explicitly prohibits it. To address this constraint, Fury Dart uses a combination of:
+
+1. Core serialization/deserialization logic
+2. Static code generation for type handling
+
+This approach ensures compatibility with Flutter while maintaining the performance and flexibility expected from Fury.
+
+## Features
+
+- XLANG mode support for cross-language serialization
+- Reference tracking for handling object graphs
+- Support for primitive types, collections, and custom classes
+- Serializer registration system
+- Code generation for class and enum serialization
+- Support for nested collections with automatic generic type conversion
+- Custom serializer registration
+- Support for using ByteWriter/ByteReader as serialization sources
+
+## Usage Examples
+
+### Basic Class Serialization
+
+```dart
+import 'package:fury/fury.dart';
+
+part 'some_class.g.dart';
```

Comment:
we could use example as package name. `some_class` gives an impression that it must be same with some_class